install-deps.sh

```diff
@@ -146,19 +146,48 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
+    ci_debug "Running clean_boost_on_ubuntu() in install-deps.sh"
+    # Find currently installed version. This may not work if there are
+    # multiple versions installed.
+    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
+                              grep -e 'libboost[0-9].[0-9]\+-dev' |
+                              cut -d' ' -f2 |
+                              cut -d'.' -f1,2)
+    # If installed_ver contains a space, we can't really count on it,
+    # but otherwise, bail out if the version installed is the version
+    # we want.
+    if test -n "$installed_ver" && echo "$installed_ver" | grep -v -q ' '; then
+        if echo "$installed_ver" | grep -q "^$boost_ver"; then
+            return
+        fi
+    fi
+
+    # Historical packages
+    $SUDO rm -f "/etc/apt/sources.list.d/ceph-libboost*.list"
+    # Currently used
+    $SUDO rm -f "/etc/apt/sources.list.d/libboost.list"
+    # Refresh package list so things aren't in the available list.
+    $SUDO apt-get update -y
+    # Remove all ceph-libboost packages. We have an early return if
+    # the desired version is already (and the only) version installed,
+    # so no need to spare it.
+    $SUDO env DEBIAN_FRONTEND=noninteractive apt-get -y remove "ceph-libboost.*"
```

Comment:
Nit: do `apt` and `apt-get` require different patterns?  Just noticing `.*` here vs `*` above when assigning `installed_ver` supposedly expressing the same thing.

Reply:
No. it tries globs first then regular expressions. But no reason not to switch it to a glob.

I'd been preferring apt-get whenever possible since apt has a warning against using it in scripts.

Reply:
But a glob or regex it can't match will error out in a way that removing an empty package list won't, so this needs a guard.